app/api/routers/token.py

```diff
@@ -767,7 +854,102 @@ def search_transfer_approval_histories(
+        )
+    if data.application_blocktimestamp_from is not None:
+        stmt = stmt.where(
+            IDXTransferApproval.application_blocktimestamp
+            >= data.application_blocktimestamp_from.astimezone(timezone.utc)
+        )
+    if data.application_blocktimestamp_to is not None:
+        stmt = stmt.where(
+            IDXTransferApproval.application_blocktimestamp
+            <= data.application_blocktimestamp_to.astimezone(timezone.utc)
+        )
+    if data.approval_datetime_from is not None:
+        stmt = stmt.where(
+            IDXTransferApproval.approval_datetime
+            >= data.approval_datetime_from.astimezone(timezone.utc)
+        )
+    if data.approval_datetime_to is not None:
+        stmt = stmt.where(
+            IDXTransferApproval.approval_datetime
+            <= data.approval_datetime_to.astimezone(timezone.utc)
+        )
+    if data.approval_blocktimestamp_from is not None:
+        stmt = stmt.where(
+            IDXTransferApproval.approval_blocktimestamp
+            >= data.approval_blocktimestamp_from.astimezone(timezone.utc)
+        )
+    if data.approval_blocktimestamp_to is not None:
+        stmt = stmt.where(
+            IDXTransferApproval.approval_blocktimestamp
+            <= data.approval_blocktimestamp_to.astimezone(timezone.utc)
+        )
+    if data.value is not None and data.value_operator is not None:
+        match data.value_operator:
+            case ValueOperator.EQUAL:
+                stmt = stmt.where(IDXTransferApproval.value == data.value)
+            case ValueOperator.GTE:
+                stmt = stmt.where(IDXTransferApproval.value >= data.value)
+            case ValueOperator.LTE:
+                stmt = stmt.where(IDXTransferApproval.value <= data.value)
+
+    count = session.scalar(select(func.count()).select_from(stmt.subquery()))
+
+    def _order(_order):
+        if _order == 0:
+            return asc
+        else:
+            return desc
+
+    if (
+        data.sort_item == "from_account_address_list"
+        and len(data.account_address_list) > 0
+    ):
+        stmt = stmt.order_by(
+            _order(data.sort_order)(
+                case(
+                    {
+                        account_address: i
+                        for i, account_address in enumerate(data.account_address_list)
+                    },
+                    value=IDXTransferApproval.from_address,
+                )
+            )
+        )
+    elif (
+        data.sort_item == "to_account_address_list"
+        and len(data.account_address_list) > 0
+    ):
+        stmt = stmt.order_by(
+            _order(data.sort_order)(
+                case(
+                    {
+                        account_address: i
+                        for i, account_address in enumerate(data.account_address_list)
+                    },
+                    value=IDXTransferApproval.to_address,
+                )
+            )
+        )
```

Comment:
Added `from_account_address_list` `to_account_address_list` to sort_item list so that the result can be ordered in arbitary order.